include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator!=(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) != 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator<(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) < 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator<=(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) <= 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator>(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) > 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
```

Comment:
There's no "we" in docs. You might say instead:

```
The function only participates in overload resolution when `s1` is convertible to `string_view`.
```

Come to think of it, `is_convertible` should be used instead of `is_constructible` (you might need to double check that)

Reply:
I believe `is_convertible` will only consider implicit conversions. I have the impression that any type convertible. implicit or explicit, should apply here.

Reply:
is_convertible only considers implicit conversions, yes. I don't see why types explicitly convertible should work here.